### ISSUE TITLE
ValueError: empty range for pyfloat

### ISSUE THREAD

**Issue by vd-indriver** [joke2k/faker#1718]
* Faker version: 14.0.0
* OS: macOS 12.5 (Monterey)

pyfloat throws ValueError when min_value and max_value have the same left digits. 

### Steps to reproduce

```
from faker import Faker

fake = Faker()

fake.pyfloat(min_value=1.25, max_value=1.75)
```

### Expected behavior

pyfloat returns a float between min_value and max_value

### Actual behavior

pyfloat throws exception

```
ValueError: empty range for randrange() (1, 1, 0)
```

**Comment by karlduggan** [comment]
Do you mind if I look into this?

**Comment by stefan6419846** [comment]
As there is no fix for this available at the moment and this appears to be a valid bug, there should not be anything which holds you back here ;)

**Comment by confuzeus** [comment]
The random_int method cast floats to int before passing to the generator. This causes 1.25 and 1.75 to turn to both turn into 1.

Since a range of 1 to 1 isn't value, a ValueError is raised. I submitted a PR to fix this at #1741.
